src/main/java/io/gravitee/policy/v3/oauth2/Oauth2PolicyV3.java

```diff
@@ -301,4 +328,31 @@ protected static boolean hasRequiredScopes(Collection<String> tokenScopes, List<
             return tokenScopes.stream().anyMatch(requiredScopes::contains);
         }
     }
+
+    protected static boolean isValidCertificateThumbprint(
+        final String tokenThumbprint,
+        final SSLSession sslSession,
+        final HttpHeaders headers,
+        final boolean ignoreMissingCnf,
+        final OAuth2PolicyConfiguration.CertificateBoundThumbprint certificateBoundThumbprint
+    ) {
+        // Ignore empty configuration method
+        if (!StringUtils.hasText(tokenThumbprint) && ignoreMissingCnf) {
+            return true;
+        } else if (!StringUtils.hasText(tokenThumbprint) && !ignoreMissingCnf) {
+            return false;
+        }
+
+        // Compute client certificate thumbprint
+        Optional<X509Certificate> clientCertificate;
+        if (certificateBoundThumbprint.isExtractCertificateFromHeader()) {
+            clientCertificate = CertificateUtils.extractCertificate(headers, certificateBoundThumbprint.getHeaderName());
+        } else {
+            clientCertificate = CertificateUtils.extractPeerCertificate(sslSession);
+        }
```

Comment:
**question**: Maybe I'm wrong but as far as I understand, this is not consistent with the behaviour described into the JIRA ticket "`When specified, the JWT policy will try to extract the client certificate using the specified HTTP header. In case of a missing header, the policy will fall back to extracting it from the current SSL session`". Here we only check the config. In addition according to the JIRA interpretation, we have to fallback to the sslSession if the "header name" is null or if the "Header value" is null.